pkg/gui/recent_repos_panel.go

```diff
@@ -1,24 +1,60 @@
 package gui
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
+	"sync"
 
 	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/lazygit/pkg/commands"
 	"github.com/jesseduffield/lazygit/pkg/env"
+	"github.com/jesseduffield/lazygit/pkg/gui/presentation/icons"
 	"github.com/jesseduffield/lazygit/pkg/gui/style"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 )
 
+func (gui *Gui) getCurrentBranch(path string) string {
+	if headFile, err := ioutil.ReadFile(fmt.Sprintf("%s/.git/HEAD", path)); err == nil {
+		content := strings.TrimSpace(string(headFile))
+		branch := strings.TrimPrefix(content, "ref: refs/heads/")
+		return branch
+	}
+	// worktrees don't have `.git/HEAD`
```

Comment:
This is just a `TODO`, I'm just curious to see if the performance is good enough